### PR TITLE
Add function for generateNgComponent

### PR DESCRIPTION
**Pull request by kawwong**

### CODE REVIEW

File: server/src/utilities/generateComponent/helpers/generateNgComponent.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        name,
+        type: 'ngComponent'
+      };
+    }
+  }
+
+  return component;
+}
+
+function containsMezzuriteDirective (filePath: string, sourceClass: ClassDeclaration): boolean {
```
Comment (manpatel3107) [inline server/src/utilities/generateComponent/helpers/generateNgComponent.ts]: Could you pls break this function into multiple.

File: server/src/utilities/generateComponent/helpers/generateNgComponent.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                    }
+                  }
+                }
+              });
+            }
+          });
+
+          if (template != null && template.length > 0) {
+            // TODO: Make this check more robust
+            containsMezzuriteDirective = template.indexOf('mezzurite') > -1;
```
Comment (manpatel3107) [inline server/src/utilities/generateComponent/helpers/generateNgComponent.ts]: Along with 'mezzurite' we need to check whether that html consists 'component-title' key value or not.

File: server/src/utilities/generateComponent/helpers/generateNgComponent.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+          } else if (templateUrl != null && templateUrl.length > 0) {
+            // In the case of both template and templateUrl are found, template is used.
+            try {
+              containsMezzuriteDirective = readFileSync(templateUrl).toString().indexOf('mezzurite') > -1;
```
Comment (manpatel3107) [inline server/src/utilities/generateComponent/helpers/generateNgComponent.ts]: If you can have this check in a separate function, you can use it foe template and templateUrl both. And also, as mentioned in a comment above we need a check for 'component-title' too.

Comment (kawwong) [reply server/src/utilities/generateComponent/helpers/generateNgComponent.ts]: Resolved via https://github.com/Microsoft/vscode-mezzurite/pull/29/commits/f05af0e98e2981a86e6f9a19b6f00ecebdba44a5.

File: server/src/utilities/generateComponent/helpers/generateNgComponent.ts (hunk 2 above)
Comment (kawwong) [reply server/src/utilities/generateComponent/helpers/generateNgComponent.ts]: Resolved via https://github.com/Microsoft/vscode-mezzurite/pull/29/commits/f05af0e98e2981a86e6f9a19b6f00ecebdba44a5.

File: server/src/utilities/generateComponent/helpers/generateNgComponent.ts (hunk 1 above)
Comment (kawwong) [reply server/src/utilities/generateComponent/helpers/generateNgComponent.ts]: Resolved via https://github.com/Microsoft/vscode-mezzurite/pull/29/commits/f05af0e98e2981a86e6f9a19b6f00ecebdba44a5.

**Pull request merged**